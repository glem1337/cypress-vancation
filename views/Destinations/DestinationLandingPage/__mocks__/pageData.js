export default {
  id: 12345,
  coordinates: {
    latitude: 40.7306,
    longitude: -73.9866,
    searchRadius: 5,
  },
  mainInfo: {
    locationName: 'New York',
    stateName: 'NY',
    stateSlug: 'new-york',
  },
  topLocations: {},
  fanFacts: {},
  nearbyDestinations: {
    isVisible: false,
  },
};
