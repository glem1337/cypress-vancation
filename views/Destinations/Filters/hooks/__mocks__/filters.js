export default {
  uuid: null,
  seats: null,
  sleeps: null,
  priceStart: null,
  priceEnd: null,
  vehicles: null,
  delivery: null,
  glamper: null,
  rating: null,
  allowPets: null,
  allowSmoking: null,
  festivalApproved: null,
  allowUnlimitedMiles: null,
  insideHeight: null,
  standardAmenities: null,
  luxuryAmenities: null,
};
