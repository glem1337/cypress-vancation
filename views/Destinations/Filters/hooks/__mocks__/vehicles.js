export default [
  {
    id: 'e966978f-b6eb-4751-bdb7-e160ed9487cf',
    name: 'Unique Camper',
    iconUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/e966978f-b6eb-4751-bdb7-e160ed9487cf/icon/eeef67069b658ab4b051a22d3a128ab4.svg',
    actualCashValue: 40000,
    description: 'Skoolies, Ambulances and Other Unique Camper Conversions.',
    estimatedEarning: 1000,
  },
  {
    id: '76418300-edf1-475a-bd3c-34ba14723a0a',
    name: 'VW Bus',
    iconUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/76418300-edf1-475a-bd3c-34ba14723a0a/icon/ee7ec84b523ec774633950b3ca441ffe.svg',
    actualCashValue: 35000,
    description: 'Iconic and Classic. Westfalias, Vanagons, Eurovans & More Volkswagen Vans.',
    estimatedEarning: 900,
  },
  {
    id: 'fa58e13e-75d1-439f-a525-c337c4b48872',
    name: 'Vehicle Camper',
    iconUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/fa58e13e-75d1-439f-a525-c337c4b48872/icon/859293aa737ffc99491fcae32786030f.svg',
    actualCashValue: 35000,
    description: 'Truck Camper Rigs. Car, Minivan, Jeep and SUV conversions, typically with roof top tents.',
    estimatedEarning: 750,
  },
  {
    id: 'def880bc-0e47-4af0-a26f-315099640df3',
    name: 'Modern Van',
    iconUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/def880bc-0e47-4af0-a26f-315099640df3/icon/36d5c37d87f3c669cee5652ec00220c9.svg',
    actualCashValue: 65000,
    description: 'Sprinters, Transits, Promasters & More Fully Built Out Camper Vans.',
    estimatedEarning: 1400,
  },
];
