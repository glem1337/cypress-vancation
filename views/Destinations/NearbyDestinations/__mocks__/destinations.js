export default [
  {
    id: '32fb2587-2dfb-40b9-9c81-584822b0ac7f',
    landingName: 'New York',
    landingSlug: 'new-york',
    landingType: 'state_landing',
    latitude: 40.73061,
    longitude: -73.935242,
    searchRadius: 4,
    stateName: null,
    stateSlug: null,
    mainPhotoUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/statelanding/32fb2587-2dfb-40b9-9c81-584822b0ac7f/main_photo/61fd9bccd8fb04fe0dfb15688f474e6b.jpg',
    mainPhotoUrl360: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/statelanding/32fb2587-2dfb-40b9-9c81-584822b0ac7f/main_photo/61fd9bccd8fb04fe0dfb15688f474e6b.jpg',
  },
];
