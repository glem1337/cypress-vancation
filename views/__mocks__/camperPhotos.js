export default [
  {
    id: 1,
    mainPhoto: true,
    photo: 'http://.../uploads/store/camperphoto',
    camper: {
      id: '0aebbaa8-5006-483b-962c-82fe01ed581f',
      status: 'draft',
    },
  },
  {
    id: 2,
    mainPhoto: false,
    photo: 'http://.../uploads/store/camperphoto',
    camper: {
      id: '0aebbaa8-5006-483b-962c-82fe01ed581f',
      status: 'draft',
    },
  },
];
