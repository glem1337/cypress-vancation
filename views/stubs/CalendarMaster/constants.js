import moment from 'moment';

export const currentDateRange = {
  from: moment('2021-04-12').startOf('day'),
  to: moment('2021-04-25').endOf('day'),
};

export const currentDate = moment('2021-04-19');

export const VANS_STATUSES = {
  published: 'Published',
  unpublished: 'Unpublished',
};

export const vansMock = [
  {
    info: {
      title: 'Adventure Ready Class B Camper Rocky Mountains',
      slug: 'super-van-145645',
      status: VANS_STATUSES.published,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-16 14:00'),
        to: moment('2021-04-19 12:00'),
        prices: [400, 500, 550],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-21 12:00'),
        prices: [400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-10 12:00'),
        to: moment('2021-04-14 14:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
  {
    info: {
      title: 'Adventure Ready Class B Camper Rocky Mountains',
      slug: 'super-van-145646',
      status: VANS_STATUSES.published,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-12 14:00'),
        to: moment('2021-04-13 12:00'),
        prices: [400],
      },
      {
        from: moment('2021-04-14 14:00'),
        to: moment('2021-04-16 12:00'),
        prices: [400, 450],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-24 12:00'),
        prices: [400, 450, 400, 400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-13 14:00'),
        to: moment('2021-04-14 20:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
  {
    info: {
      title: 'Adventure Ready Class 2',
      slug: 'super-van-145647',
      status: VANS_STATUSES.published,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-16 14:00'),
        to: moment('2021-04-19 12:00'),
        prices: [400, 500, 550],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-21 12:00'),
        prices: [400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-17 14:00'),
        to: moment('2021-04-20 12:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
  {
    info: {
      title: 'Adventure Ready Class 2',
      slug: 'super-van-145648',
      status: VANS_STATUSES.unpublished,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-12 14:00'),
        to: moment('2021-04-13 12:00'),
        prices: [400],
      },
      {
        from: moment('2021-04-14 14:00'),
        to: moment('2021-04-16 12:00'),
        prices: [400, 450],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-24 12:00'),
        prices: [400, 450, 400, 400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-11 16:00'),
        to: moment('2021-04-28 12:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
  {
    info: {
      title: 'Adventure Ready Class 2',
      slug: 'super-van-145649',
      status: VANS_STATUSES.unpublished,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-16 14:00'),
        to: moment('2021-04-19 12:00'),
        prices: [400, 500, 550],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-21 12:00'),
        prices: [400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-23 16:00'),
        to: moment('2021-04-27 12:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
];
