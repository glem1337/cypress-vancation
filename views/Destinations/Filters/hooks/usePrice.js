import { useState, useEffect, useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import * as R from 'ramda';
import { Input, Slider } from 'antd';
import { FormattedMessage } from 'react-intl';
import { debounce } from 'lodash';

import { FILTERS } from 'constants/searchDestinations';
import { searchDestinationFiltersSelector } from 'state/concepts/search-destinations/selectors';
import { setFilterBatchValue } from 'state/concepts/search-destinations/actions';

import useSharedValues from './useSharedValues';

const REGEX = /[^0-9]+/g;

function useContainer(props) {
  const dispatch = useDispatch();

  const persistedFilters = useSelector(searchDestinationFiltersSelector);

  // Shared values.
  const { tipFormatter } = useSharedValues();

  /**
   * Prepare initial value.
   */
  const prepareInitialValue = (filterName) => {
    const filter = R.path([filterName], persistedFilters);

    if (R.isNil(filter) && filterName === 'priceStart') {
      return `$${FILTERS.PRICE_BOUNDARIES.MIN}`;
    }

    if (!R.isNil(filter) && filterName === 'priceStart' && filter >= FILTERS.PRICE_BOUNDARIES.MAX) {
      return `$${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    if (R.isNil(filter) && filterName === 'priceEnd') {
      return `$${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    if (!R.isNil(filter) && filterName === 'priceEnd' && filter >= FILTERS.PRICE_BOUNDARIES.MAX) {
      return `$${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    return `$${filter}`;
  };

  /**
   * Use state.
   */
  const [filters, setFilters] = useState({
    priceStart: prepareInitialValue('priceStart'),
    priceEnd: prepareInitialValue('priceEnd'),
  });

  /**
   * Clear filters.
   */
  const clearFilters = () => {
    dispatch(setFilterBatchValue([
      {
        name: 'priceStart',
        value: null,
      },
      {
        name: 'priceEnd',
        value: null,
      },
    ]));

    if (props.toggleOpenedState) {
      props.toggleOpenedState(null)();
    }
  };

  /**
   * Apply filters.
   */
  const applyFilters = (params = {}) => {
    // Prepare start price
    let priceStart = (params.priceStart || filters.priceStart).replace(REGEX, '');
    priceStart = parseInt(priceStart, 10);

    // Prepare end price
    let priceEnd = (params.priceEnd || filters.priceEnd).replace(REGEX, '');
    priceEnd = parseInt(priceEnd, 10);

    if (priceStart === FILTERS.PRICE_BOUNDARIES.MIN && priceEnd === FILTERS.PRICE_BOUNDARIES.MAX) {
      priceStart = null;
      priceEnd = null;
    }

    dispatch(setFilterBatchValue([
      {
        name: 'priceStart',
        value: priceStart,
      },
      {
        name: 'priceEnd',
        value: priceEnd,
      },
    ]));

    if (props.toggleOpenedState) {
      props.toggleOpenedState(null)();
    }

    return {
      priceStart,
      priceEnd,
    };
  };

  /**
   * Persisted handlers.
   */
  const handlers = useRef({
    /**
     * Immediately Apply filters.
     */
    applyFiltersImmediately: debounce(applyFilters, FILTERS.DEBOUNCE_TIME),
  });

  /**
   * Set range value
   */
  const onChangePriceBySlider = (range) => {
    let priceStart = R.head(range);
    let priceEnd = R.last(range);

    if (priceStart >= FILTERS.PRICE_BOUNDARIES.MAX) {
      priceStart = `$${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    } else {
      priceStart = `$${priceStart}`;
    }

    if (priceEnd >= FILTERS.PRICE_BOUNDARIES.MAX) {
      priceEnd = `$${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    } else {
      priceEnd = `$${priceEnd}`;
    }

    setFilters({
      priceStart,
      priceEnd,
    });

    if (props.immediatelyApply) {
      handlers.current.applyFiltersImmediately({
        priceStart,
        priceEnd,
      });
    }

    return {
      priceStart,
      priceEnd,
    };
  };

  /**
   * On change price
   */
  const onChangePrice = (filterName) => ({ target: { value } }) => {
    let formatted = value;

    formatted = value.replace(REGEX, '');

    if (parseInt(formatted, 10) > FILTERS.PRICE_BOUNDARIES.MAX) {
      formatted = `${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    setFilters(prev => ({
      ...prev,
      [filterName]: `$${formatted}`,
    }));

    if (props.immediatelyApply) {
      handlers.current.applyFiltersImmediately({
        ...filters,
        [filterName]: formatted,
      });
    }
  };

  /**
   * Normalize value.
   */
  const rangeValue = () => {
    const { priceStart, priceEnd } = filters;

    let start = priceStart.replace(REGEX, '');
    start = parseInt(start, 10);

    let end = priceEnd.replace(REGEX, '');
    end = parseInt(end, 10);

    if (Number.isNaN(start)) {
      start = 0;
    }

    if (Number.isNaN(end)) {
      end = 0;
    }

    if (start > FILTERS.PRICE_BOUNDARIES.MAX) {
      start = FILTERS.PRICE_BOUNDARIES.MAX;
    }

    if (end > FILTERS.PRICE_BOUNDARIES.MAX) {
      end = FILTERS.PRICE_BOUNDARIES.MAX;
    }

    return [start, end];
  };

  /**
   * On blur handler
   */
  const onBlurHandler = (filterName) => () => {
    let filter = R.prop(filterName, filters);

    filter = filter.replace(REGEX, '');

    filter = parseInt(filter, 10);

    if (filterName === 'priceStart' && Number.isNaN(filter)) {
      filter = `$${FILTERS.PRICE_BOUNDARIES.MIN}`;
    }

    if (filterName === 'priceEnd' && Number.isNaN(filter)) {
      filter = `$${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    if (filter >= FILTERS.PRICE_BOUNDARIES.MAX) {
      filter = `$${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    const value = typeof filter === 'number'
      ? `$${filter}`
      : filter;

    setFilters(prev => ({
      ...prev,
      [filterName]: value,
    }));

    return true;
  };

  /**
   * Is filter applied.
   */
  const getPriceString = () => {
    if (R.isNil(persistedFilters.priceStart) && R.isNil(persistedFilters.priceEnd)) {
      return null;
    }

    if (
      persistedFilters.priceStart === FILTERS.PRICE_BOUNDARIES.MIN
      && persistedFilters.priceEnd === FILTERS.PRICE_BOUNDARIES.MIN
    ) {
      return null;
    }

    if (
      persistedFilters.priceStart === FILTERS.PRICE_BOUNDARIES.MIN
      && persistedFilters.priceEnd === FILTERS.PRICE_BOUNDARIES.MAX
    ) {
      return null;
    }

    if (
      persistedFilters.priceStart >= FILTERS.PRICE_BOUNDARIES.MAX
      && persistedFilters.priceEnd >= FILTERS.PRICE_BOUNDARIES.MAX
    ) {
      return `$${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    let priceStart = R.prop('priceStart', persistedFilters);
    let priceEnd = R.prop('priceEnd', persistedFilters);

    if (priceStart >= FILTERS.PRICE_BOUNDARIES.MAX) {
      priceStart = `${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    if (priceEnd >= FILTERS.PRICE_BOUNDARIES.MAX) {
      priceEnd = `${FILTERS.PRICE_BOUNDARIES.MAX}+`;
    }

    return `$${priceStart} - $${priceEnd}`;
  };

  /**
   * Render price widget
   */
  const renderPriceWidget = () => (
    <>
      <div className="d-flex align-items-center justify-content-space-between mb-24 w-280">
        <div className="w-120">
          <div className="font-12 mb-4">
            <FormattedMessage id="shared.from" />
          </div>
          <Input
            className="mb-0"
            autoComplete="off"
            onChange={onChangePrice('priceStart')}
            value={filters.priceStart}
            onBlur={onBlurHandler('priceStart')}
          />
        </div>
        <div className="search-page__filters-popover__line" />
        <div className="w-120">
          <div className="font-12 mb-4">
            <FormattedMessage id="shared.to" />
          </div>
          <Input
            className="mb-0"
            autoComplete="off"
            onChange={onChangePrice('priceEnd')}
            value={filters.priceEnd}
            onBlur={onBlurHandler('priceEnd')}
          />
        </div>
      </div>
      <Slider
        value={rangeValue()}
        max={FILTERS.PRICE_BOUNDARIES.MAX}
        range
        tipFormatter={tipFormatter(FILTERS.PRICE_BOUNDARIES.MAX)}
        onChange={onChangePriceBySlider}
      />
    </>
  );

  /**
   * Check filter updates
   */
  const checkFilterUpdates = () => {
    if (!persistedFilters.uuid) {
      return false;
    }

    setFilters({
      priceStart: prepareInitialValue('priceStart'),
      priceEnd: prepareInitialValue('priceEnd'),
    });

    return true;
  };

  /**
   * Updating
   */
  useEffect(() => {
    checkFilterUpdates();
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [persistedFilters.uuid]);

  /**
   * Updating
   */
  useEffect(() => {
    checkFilterUpdates();
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.openedFilterName]);

  return {
    checkFilterUpdates,
    prepareInitialValue,
    onChangePriceBySlider,
    clearFilters,
    applyFilters,
    priceString: getPriceString(),
    filters,
    onChangePrice,
    rangeValue: rangeValue(),
    onBlurHandler,
    renderPriceWidget,
    setFilters,
  };
}

export default useContainer;
