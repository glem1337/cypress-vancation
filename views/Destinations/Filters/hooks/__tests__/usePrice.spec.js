import { act, renderHook } from '@testing-library/react-hooks';

import { dispatch } from '__mocks__/react-redux';
import { setFilterBatchValue } from 'state/concepts/search-destinations/actions';
import { FILTERS } from 'constants/searchDestinations';
import { searchDestinationFiltersSelector } from 'state/concepts/search-destinations/selectors';

import usePrice from '../usePrice';
import mockedFilter from '../__mocks__/filters';

jest.mock('lodash/debounce', () => fn => fn);

jest.mock('state/concepts/search-destinations/selectors', () => ({
  searchDestinationFiltersSelector: jest.fn(() => mockedFilter),
}));

const mockedHookData = {
  tipFormatter: jest.fn(),
};
jest.mock('views/Destinations/Filters/hooks/useSharedValues', () => jest.fn(() => mockedHookData));

describe('usePrice hook', () => {
  let result = null;

  const props = {
    toggleOpenedState: jest.fn(() => jest.fn()),
  };

  beforeEach(() => {
    searchDestinationFiltersSelector.mockReturnValue(mockedFilter);

    ({ result } = renderHook(() => usePrice(props)));

    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  it('checks `clearFilters` method', () => {
    act(() => {
      result.current.clearFilters();
    });

    expect(dispatch).toHaveBeenCalledWith(setFilterBatchValue([
      {
        name: 'priceStart',
        value: null,
      },
      {
        name: 'priceEnd',
        value: null,
      },
    ]));

    expect(props.toggleOpenedState).toHaveBeenCalled();
  });

  it('checks `applyFilters` method', () => {
    let priceStart = null;
    let priceEnd = null;

    act(() => {
      ({ priceStart, priceEnd } = result.current.applyFilters());
    });

    expect(dispatch).toHaveBeenCalledWith(setFilterBatchValue([
      {
        name: 'priceStart',
        value: priceStart,
      },
      {
        name: 'priceEnd',
        value: priceEnd,
      },
    ]));

    expect(props.toggleOpenedState).toHaveBeenCalled();
  });

  describe('checks `prepareInitialValue` method', () => {
    it('for start price when null', () => {
      let res = null;

      act(() => {
        res = result.current.prepareInitialValue('priceStart');
      });

      expect(res).toBe(`$${FILTERS.PRICE_BOUNDARIES.MIN}`);
    });

    it('for end price when null', () => {
      let res = null;

      act(() => {
        res = result.current.prepareInitialValue('priceEnd');
      });

      expect(res).toBe(`$${FILTERS.PRICE_BOUNDARIES.MAX}+`);
    });

    it('for start price when equals max', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: FILTERS.PRICE_BOUNDARIES.MAX,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.prepareInitialValue('priceStart');
      });

      expect(res).toBe(`$${FILTERS.PRICE_BOUNDARIES.MAX}+`);
    });

    it('for end price when equals max', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceEnd: FILTERS.PRICE_BOUNDARIES.MAX,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.prepareInitialValue('priceEnd');
      });

      expect(res).toBe(`$${FILTERS.PRICE_BOUNDARIES.MAX}+`);
    });

    it('for end price when does not equals max', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceEnd: 111,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.prepareInitialValue('priceEnd');
      });

      expect(res).toBe(`$${111}`);
    });
  });

  describe('checks `onChangePriceBySlider` method', () => {
    it('for not boundaries', () => {
      const range = [12, 33];

      act(() => {
        result.current.onChangePriceBySlider(range);
      });

      expect(result.current.filters).toEqual({
        priceStart: '$12',
        priceEnd: '$33',
      });
    });

    it('for boundaries', () => {
      const range = [
        FILTERS.PRICE_BOUNDARIES.MAX,
        FILTERS.PRICE_BOUNDARIES.MAX,
      ];

      act(() => {
        result.current.onChangePriceBySlider(range);
      });

      expect(result.current.filters).toEqual({
        priceStart: `$${FILTERS.PRICE_BOUNDARIES.MAX}+`,
        priceEnd: `$${FILTERS.PRICE_BOUNDARIES.MAX}+`,
      });
    });

    it('with immediate update', () => {
      ({ result } = renderHook(() => usePrice({
        ...props,
        immediatelyApply: true,
      })));

      const range = [12, 33];

      act(() => {
        result.current.onChangePriceBySlider(range);
      });

      expect(result.current.filters).toEqual({
        priceStart: '$12',
        priceEnd: '$33',
      });

      expect(dispatch).toHaveBeenCalledWith(setFilterBatchValue([
        {
          name: 'priceStart',
          value: 12,
        },
        {
          name: 'priceEnd',
          value: 33,
        },
      ]));
    });
  });

  describe('checks `onChangePrice` method', () => {
    it('for not boundaries', () => {
      const event = {
        target: {
          value: '123',
        },
      };

      act(() => {
        result.current.onChangePrice('priceStart')(event);
      });

      expect(result.current.filters).toEqual({
        ...result.current.filters,
        priceStart: '$123',
      });
    });

    it('for boundaries', () => {
      const event = {
        target: {
          value: `${FILTERS.PRICE_BOUNDARIES.MAX + 1}`,
        },
      };

      act(() => {
        result.current.onChangePrice('priceStart')(event);
      });

      expect(result.current.filters).toEqual({
        ...result.current.filters,
        priceStart: `$${FILTERS.PRICE_BOUNDARIES.MAX}+`,
      });
    });

    it('with immediate update', () => {
      const event = {
        target: {
          value: '123',
        },
      };

      ({ result } = renderHook(() => usePrice({
        ...props,
        immediatelyApply: true,
      })));

      act(() => {
        result.current.onChangePrice('priceStart')(event);
      });

      expect(dispatch).toHaveBeenCalledWith(setFilterBatchValue([
        {
          name: 'priceStart',
          value: 123,
        },
        {
          name: 'priceEnd',
          value: 500,
        },
      ]));
    });
  });

  describe('checks `rangeValue` method', () => {
    it('should return zeros', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: 'test',
        priceEnd: 'test',
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = 0;

      act(() => {
        res = result.current.rangeValue;
      });

      expect(res).toEqual([0, 0]);
    });

    it('should boundaries', () => {
      act(() => {
        result.current.setFilters({
          priceStart: `${FILTERS.PRICE_BOUNDARIES.MAX + 100}`,
          priceEnd: `${FILTERS.PRICE_BOUNDARIES.MAX + 100}`,
        });
      });

      let res = 0;

      act(() => {
        res = result.current.rangeValue;
      });

      expect(res).toEqual([FILTERS.PRICE_BOUNDARIES.MAX, FILTERS.PRICE_BOUNDARIES.MAX]);
    });

    it('should simple values', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: 1,
        priceEnd: 2,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = 0;

      act(() => {
        res = result.current.rangeValue;
      });

      expect(res).toEqual([1, 2]);
    });
  });

  describe('checks `onBlurHandler` method', () => {
    it('should return min for priceStart', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: 'test',
        priceEnd: 'test',
      });

      ({ result } = renderHook(() => usePrice(props)));

      act(() => {
        result.current.onBlurHandler('priceStart')();
      });

      expect(result.current.filters.priceStart).toBe(`$${FILTERS.PRICE_BOUNDARIES.MIN}`);
    });

    it('should return max for priceEnd', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: 'test',
        priceEnd: 'test',
      });

      ({ result } = renderHook(() => usePrice(props)));

      act(() => {
        result.current.onBlurHandler('priceEnd')();
      });

      expect(result.current.filters.priceEnd).toBe(`$${FILTERS.PRICE_BOUNDARIES.MAX}+`);
    });

    it('should return max for big value', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: 'test',
        priceEnd: 10000,
      });

      ({ result } = renderHook(() => usePrice(props)));

      act(() => {
        result.current.onBlurHandler('priceEnd')();
      });

      expect(result.current.filters.priceEnd).toBe(`$${FILTERS.PRICE_BOUNDARIES.MAX}+`);
    });
  });

  describe('checks `getPriceString` method', () => {
    it('should return null', () => {
      let res = null;

      act(() => {
        res = result.current.priceString;
      });

      expect(res).toBe(null);
    });

    it('should return null', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: FILTERS.PRICE_BOUNDARIES.MIN,
        priceEnd: FILTERS.PRICE_BOUNDARIES.MIN,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.priceString;
      });

      expect(res).toBe(null);
    });

    it('should return null', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: FILTERS.PRICE_BOUNDARIES.MIN,
        priceEnd: FILTERS.PRICE_BOUNDARIES.MAX,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.priceString;
      });

      expect(res).toBe(null);
    });

    it('should return max single value', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: FILTERS.PRICE_BOUNDARIES.MAX,
        priceEnd: FILTERS.PRICE_BOUNDARIES.MAX,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.priceString;
      });

      expect(res).toBe('$500+');
    });

    it('should return value when start price more than max', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: FILTERS.PRICE_BOUNDARIES.MAX + 1,
        priceEnd: 111,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.priceString;
      });

      expect(res).toBe('$500+ - $111');
    });

    it('should return max value', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: FILTERS.PRICE_BOUNDARIES.MAX + 1,
        priceEnd: FILTERS.PRICE_BOUNDARIES.MAX + 1,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.priceString;
      });

      expect(res).toBe(`$${FILTERS.PRICE_BOUNDARIES.MAX}+`);
    });

    it('should return max value', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: FILTERS.PRICE_BOUNDARIES.MAX,
        priceEnd: FILTERS.PRICE_BOUNDARIES.MAX,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.priceString;
      });

      expect(res).toBe(`$${FILTERS.PRICE_BOUNDARIES.MAX}+`);
    });

    it('should return range string', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        priceStart: 11,
        priceEnd: 22,
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.priceString;
      });

      expect(res).toBe('$11 - $22');
    });
  });

  it('checks `renderPriceWidget` method', () => {
    let res = null;

    act(() => {
      res = result.current.renderPriceWidget();
    });

    expect(res).toMatchSnapshot();
  });

  describe('checks `checkFilterUpdates` method', () => {
    it('should return false', () => {
      let res = null;

      act(() => {
        res = result.current.checkFilterUpdates();
      });

      expect(res).toBe(false);
    });

    it('should return true', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        uuid: '213218n-vdjiks',
      });

      ({ result } = renderHook(() => usePrice(props)));

      let res = null;

      act(() => {
        res = result.current.checkFilterUpdates();
      });

      expect(res).toBe(true);
    });
  });
});
