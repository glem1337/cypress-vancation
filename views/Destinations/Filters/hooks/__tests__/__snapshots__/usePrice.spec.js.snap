// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePrice hook checks \`renderPriceWidget\` method 1`] = `
<React.Fragment>
  <div
    className="d-flex align-items-center justify-content-space-between mb-24 w-280"
  >
    <div
      className="w-120"
    >
      <div
        className="font-12 mb-4"
      >
        <Memo(FormattedMessage)
          id="shared.from"
        />
      </div>
      <Input
        autoComplete="off"
        className="mb-0"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="$0"
      />
    </div>
    <div
      className="search-page__filters-popover__line"
    />
    <div
      className="w-120"
    >
      <div
        className="font-12 mb-4"
      >
        <Memo(FormattedMessage)
          id="shared.to"
        />
      </div>
      <Input
        autoComplete="off"
        className="mb-0"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="$500+"
      />
    </div>
  </div>
  <Slider
    max={500}
    onChange={[Function]}
    range={true}
    tipFormatter={[Function]}
    value={
      Array [
        0,
        500,
      ]
    }
  />
</React.Fragment>
`;

exports[`usePrice hook matches snapshot 1`] = `
Object {
  "applyFilters": [Function],
  "checkFilterUpdates": [Function],
  "clearFilters": [Function],
  "filters": Object {
    "priceEnd": "$500+",
    "priceStart": "$0",
  },
  "onBlurHandler": [Function],
  "onChangePrice": [Function],
  "onChangePriceBySlider": [Function],
  "prepareInitialValue": [Function],
  "priceString": null,
  "rangeValue": Array [
    0,
    500,
  ],
  "renderPriceWidget": [Function],
  "setFilters": [Function],
}
`;
